src/settings/BatchGroupConfigurationSettings/BatchGroupConfigurationSettings.js

```diff
@@ -31,14 +43,87 @@ const BatchGroupConfigurationSettings = ({ mutator }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   []);
 
-  const onSaveExportConfig = useCallback(
-    () => {
+  const fetchExportConfig = useCallback(
+    (id) => {
+      setIsLoading(true);
+      setExportConfig();
+      setCredentials();
 
+      mutator.exportConfig.GET({
+        params: {
+          query: `batchGroupId==${id}`,
+        },
+      })
+        .then(exportConfigResp => {
+          if (exportConfigResp.length) {
+            const exportConfigId = exportConfigResp[0].id;
+
+            setExportConfig(exportConfigResp[0]);
+            mutator.exportConfigId.update({ id: exportConfigId });
+
+            return mutator.credentials.GET();
+          } else {
+            setExportConfig({});
+
+            return Promise.resolve();
+          }
+        })
+        .then(credentialsResp => {
+          if (credentialsResp) setCredentials(credentialsResp);
+          else setCredentials({});
+        })
+        .finally(() => setIsLoading(false));
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [selectedBatchGroupId],
   );
 
+  useEffect(() => {
+    fetchExportConfig(selectedBatchGroupId);
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [selectedBatchGroupId]);
+
+  const onSave = useCallback(
+    (formValues) => {
+      saveExportConfig(formValues, mutator, credentials)
+        .then(() => {
+          fetchExportConfig(selectedBatchGroupId);
+          showCallout({
+            messageId: 'ui-invoice.settings.batchGroupConfiguration.save.success',
+            type: 'success',
+          });
+        })
+        .catch(() => showCallout({
+          messageId: 'ui-invoice.settings.batchGroupConfiguration.save.error',
+          type: 'error',
+        }));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedBatchGroupId, credentials],
+  );
+
+  const selectBatchGroup = useCallback(
```

Comment:
why you can't use `setSelectedBatchGroupId` directly? it's looks the same as wrapper, what is the purpose of wrapper then?

Reply:
It's possible, but in this case, a warning appears in the console

Reply:
could you please provide that warning? I can't imagine of any

Reply:
Checked it again - no warning. So warning that appeared was fixed and had another reason